shelley/chain-and-ledger/executable-spec/src/STS/Ledgers.hs

```diff
@@ -25,15 +25,16 @@ import           PParams
 import           Slot
 import           Tx
 import           Updates (Applications (..), UpdateState (..), apps, newAVs)
+import           UTxO (UTxO (..))
 
 import           Control.State.Transition
 
 import           STS.Ledger
 
 data LEDGERS hashAlgo dsignAlgo
 
-data LedgersEnv
-  = LedgersEnv Slot PParams
+data LedgersEnv hashAlgo dsignAlgo
+  = LedgersEnv (UTxO hashAlgo dsignAlgo) Slot PParams
```

Comment:
why is the UTxO part of the LEDGERS env? The state contains the LedgerState which contains the UTxO

Reply:
yes you're right, I took a a wrong turn here, I've reversed this (and you'll see a new HasTrace function `traceOfLengthWithInitState` that allows for passing in the initial state when building a trace